src/Nether.Analytics/MessagePipeline.cs

```diff
@@ -38,11 +39,20 @@ public class MessagePipeline
                 }
             }
 
+
             //TODO: Run output managers in parallel
             var outputManagerIdx = 0;
             foreach (var outputManager in _outputManagers)
             {
-                await outputManager.OutputMessageAsync(PipelineName, outputManagerIdx++, msg);
+                await outputManager.OutputMessageAsync(partitionId, PipelineName, outputManagerIdx++, msg);
+            }
+        }
+
+        public async Task FlushAsync(string partitionId)
+        {
+            foreach (var outputManager in _outputManagers)
+            {
+                await outputManager.FlushAsync(partitionId);
```

Comment:
Does it make sense to flush the partitions in parallel? I.e. not await on each FlushAsync call but track the returned tasks and WaitAll?